pkg/controller/certificaterequests/sync.go

```diff
@@ -158,14 +158,14 @@ func (c *Controller) updateCertificateRequestStatusAndAnnotations(ctx context.Co
 	log := logf.FromContext(ctx, "updateStatus")
 
 	// if annotations changed we have to call .Update() and not .UpdateStatus()
-	if !reflect.DeepEqual(old.Annotations, new.Annotations) {
+	if !apiequality.Semantic.DeepEqual(old.Annotations, new.Annotations) {
 		log.V(logf.DebugLevel).Info("updating resource due to change in annotations", "diff", pretty.Diff(old.Annotations, new.Annotations))
 		return c.cmClient.CertmanagerV1().CertificateRequests(new.Namespace).Update(context.TODO(), new, metav1.UpdateOptions{})
 	}
 
 	oldBytes, _ := json.Marshal(old.Status)
```

Comment:
I think that here we were originally marshalling the status because of issues with  `reflect.DeepEqual`  (as I understand from @munnerz 's comment [here](https://github.com/jetstack/cert-manager/pull/3525#issuecomment-748018717))

So perhaps, now we could just use semantic check to compare the old and new status directly, without marshalling?

Reply:
Yes, I also gather that it can be replaced.
So shall I replace `oldBytes` and `newbytes` [here](https://github.com/jetstack/cert-manager/blob/master/pkg/controller/certificaterequests/sync.go#L172) also, as from what I could understand `pretty.Diff` should work fine with `cmapi.CertificateRequestStatus` instances as its argument, but I am not sure. It would be great if you could point me in the right direction.

Reply:
Yes, I think it would be okay to use `pretty.Diff` with statuses directly and get rid of the `oldBytes`, `newBytes` there as well.

We use `pretty.Diff` to print parts of custom resources in [other places](https://github.com/jetstack/cert-manager/blob/master/pkg/controller/acmeorders/util_test.go#L1272) too and the output looks ok. 
Worst case, there would be an unexpected timestamp diff included, but I think that shouldn't be a problem?

Reply:
Yes, I think it should be fine. I have made the suggested changes.